scheduler/server/stateful_scheduler.go

```diff
@@ -1162,6 +1146,29 @@ func (s *statefulScheduler) processKillJobRequests(reqs []jobKillRequest) {
 	}
 }
 
+// abortTask abort a task - will be triggered by killing a job and when the scheduling algorithm rebalances the workers
```

Comment:
Is this substantially different than the existing abort task mechanisms? Why did we need it?

Reply:
no - it's copying the abort functionality from inline in processKillJobRequests() and exposing it so it can be used by both processKillJobRequests(), and the taskScheduler.go getTaskAssignments when stopping tasks for rebalancing.